### PR TITLE
Add Besu Plugin information

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst <byron.gravenorst@consensys.net>

## PR description
 Add information about the Besu Plugin API and link to information about how to build a Plugin.

### CODE REVIEW

File: docs/Concepts/Plugins.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    level below (`../`) the besu executable.
+
+Each plugin in the directory has the following lifecycle events:
+
+* **Register** - Executed when Besu starts. Besu checks plugin compatibility and registers plugins.
+* **Start** - Plugins are started after being successfully registered.
+* **Stop** - Plugins are stopped by Besu.
+
+!!! note
+    Besu does not guarantee the order of the lifecycle events.
```
Comment (shemnon) [inline docs/Concepts/Plugins.md]: We do guarantee register->start->stop, just not which plugin goes first.
```suggestion
    Besu does not guarantee the order plugins will be called during lifecycle events.
```

File: docs/Reference/Plugin-API-Interfaces.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+description: Plugin interfaces
+<!--- END of page meta data -->
+
+Hyperledger Besu provides API interfaces to allow users to [build plugins](../Concepts/Plugins.md) to
+extend Besu functionality.
+
+
+
+| Interface          | Description                                              |
+|--------------------|----------------------------------------------------------|
```
Comment (shemnon) [inline docs/Reference/Plugin-API-Interfaces.md]: I would split this out into two tables.  BesuContext and BesuPlugin provide the core plugin classes.  All the other classes are services that can be retrieved.

This list is missing some services
**BesuConfiguration** - provides the file system locations of Besu's storage
**MetricCategoryRegistry** - adds a new metrics category to the CLI
**MetricsSystem** - Allows you to register metrics with the prometheus endpoint
**PoAQueryService** - For Clique and IBFT networks lets you query the current state of consensus
**IbftQueryService** - Like PoAQueryService but with extra IBFT specific queries.

in addition to PicoCLIOptions, BesuEvents, and StorageService

File: docs/Concepts/Plugins.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Concepts/Plugins.md]: Done.

File: docs/Reference/Plugin-API-Interfaces.md (hunk 2 above)
Comment (bgravenorst) [reply docs/Reference/Plugin-API-Interfaces.md]: Done.

File: docs/Concepts/Plugins.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+description: Plugins
+<!--- END of page meta data -->
+
+# Plugins
+
+You can extend Hyperledger Besu functionality by building Java plugins. The
+Plugin API allows you to take data from any Besu network, public or
+permissioned, and feed it into an application or system.
```
Comment (MadelineMurray) [inline docs/Concepts/Plugins.md]: The Plugin API allows you to -> Use the Plugin API 

Suggested edit to make active

File: docs/Concepts/Plugins.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+<!--- END of page meta data -->
+
+# Plugins
+
+You can extend Hyperledger Besu functionality by building Java plugins. The
+Plugin API allows you to take data from any Besu network, public or
+permissioned, and feed it into an application or system. 
+
+For example, a plugin can add additional monitoring functionality or stream
+event data to a third-party application. The API exposes data about the
```
Comment (MadelineMurray) [inline docs/Concepts/Plugins.md]: For example, a plugin can add additional  -> For example, create a plugin to add additional 

Suggested edit to make active

File: docs/Concepts/Plugins.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    View the [Plugin API webinar](https://youtu.be/78sa2WuA1rg) for an example of how to build a plugin.
+
+## Install Plugins
+
+The plugin (`.jar`) must be copied to the `plugins` directory to allow Besu 
+access and use the plugin.
+
+!!! important
+    If not already present, the `plugins` directory must be created one directory
+    level below (`../`) the besu executable.
```
Comment (MadelineMurray) [inline docs/Concepts/Plugins.md]: besu executable -> `besu` executable 

Without the code style I wondered why besu wasn't Besu

File: docs/Reference/Plugin-API-Interfaces.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+description: Plugin interfaces
+<!--- END of page meta data -->
+
+# Plugin API Interfaces
+
+Hyperledger Besu provides API interfaces to allow users to [build plugins](../Concepts/Plugins.md) to
```
Comment (MadelineMurray) [inline docs/Reference/Plugin-API-Interfaces.md]: Hyperledger Besu provides API interfaces to allow users -> API interfaces in Hyperledger Besu allow users 

Suggested edit to make more active

File: docs/Reference/Plugin-API-Interfaces.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+The following interfaces provide services that can be retrieved.
+
+| Interface          | Description                                              |
+|--------------------|----------------------------------------------------------|
+| **BesuEvents**     | Allows plugins to attach to various events during Besu operation.  |
+| **BesuConfiguration** | Provides file system locations of Besu's storage. |
+| **IbftQueryService** | Allows IBFT 2.0 aspects of the blockchain to be queried. |
+| **MetricCategoryRegistry** | Adds a new metrics category to the CLI. |
+| **MetricsSystem**  | Register metrics with the Prometheus endpoint. |
+| **PoAQueryService** | Query the current state of Clique and IBFT consensus protocols. |
```
Comment (MadelineMurray) [inline docs/Reference/Plugin-API-Interfaces.md]: IBFT -> IBFT 2.0

File: docs/Concepts/Plugins.md (hunk 3 above)
Comment (bgravenorst) [reply docs/Concepts/Plugins.md]: Done.

File: docs/Concepts/Plugins.md (hunk 4 above)
Comment (bgravenorst) [reply docs/Concepts/Plugins.md]: Done.

File: docs/Concepts/Plugins.md (hunk 5 above)
Comment (bgravenorst) [reply docs/Concepts/Plugins.md]: Done.

File: docs/Reference/Plugin-API-Interfaces.md (hunk 6 above)
Comment (bgravenorst) [reply docs/Reference/Plugin-API-Interfaces.md]: Done

File: docs/Reference/Plugin-API-Interfaces.md (hunk 7 above)
Comment (bgravenorst) [reply docs/Reference/Plugin-API-Interfaces.md]: Done.

**Pull request merged**